### PR TITLE
rebase solaris support

### PR DESCRIPTION
**Pull request by Phil-Friderici**
to get https://github.com/ghoneycutt/puppet-module-nscd/pull/19 done before it have it's first birthday

First commit contains the old ```init_spec.rb``` to prove the patch is backward compatibility.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -94,7 +274,6 @@
   }
   validate_bool($service_enable_real)
 
-  validate_absolute_path($logfile)
```
Comment (Phil-Friderici) [inline manifests/init.pp]: went down to [line 523 ](https://github.com/ghoneycutt/puppet-module-nscd/pull/36/files#diff-60ae41fd0a31977447947f59940ee9a4R523)

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -237,7 +237,7 @@
       it 'should fail' do
         expect {
           should contain_class('nscd')
-        }.to raise_error(Puppet::Error,/nscd supports osfamilies Debian, RedHat and Suse. Detected osfamily is <unsupported>\./)
+        }.to raise_error(Puppet::Error,/Nscd supports osfamilies Debian, RedHat, Suse and Solaris\. Detected osfamily is <unsupported>\./)
```
Comment (Phil-Friderici) [inline spec/classes/init_spec.rb]: This is the only (non functional) change to rspec to reflect new error message containint Solaris support.

**Comment by Phil-Friderici** [comment]
The second commit contains the spec tests for the new functionality

**Comment by Phil-Friderici** [comment]
too bad Puppet v4 still throughs errors :(

**Comment by ghoneycutt** [comment]
This seems to be issues with the error messages that change in Puppet v4. I think you can just remove the carrots `^` from the error matching in the spec tests and this should pass.

**Comment by Phil-Friderici** [comment]
Exactly, they sneaked back in with Solaris specific tests. Fixed it, should be passing now.

**Comment by Phil-Friderici** [comment]
here we go, all back to green :)
@ghoneycutt have a nice weekend !

**Comment by ghoneycutt** [comment]
@Phil-Friderici Looks good. Could you use in your infrastructure on Solaris 10 and non-Solaris 10 systems to be sure we didn't miss anything?

**Comment by ghoneycutt** [comment]
Guten Feierabend! :)

**Comment by Phil-Friderici** [comment]
Will verify/put it in production on monday morning. My local Solaris specialist is looking forward to give this a try :)

**Comment by Phil-Friderici** [comment]
...still waiting for real life verification...

**Comment by ghoneycutt** [comment]
@Phil-Friderici No prob. When you say it's good, I'll merge :)

**Comment by Phil-Friderici** [comment]
Real life testing showed that the source commit didn't took care for the Solaris specific package parameters. Adding/testing these at the very moment.

**Comment by Phil-Friderici** [comment]
Now it works, but we need another rebase against master. As travis also stopped testing here, I'll create a new PR for that.

**Pull request merged**